src/timely-util/src/order.rs

```diff
@@ -168,6 +209,35 @@ impl<P: Eq> Interval<P> {
     }
 }
 
+impl<P: PartialOrd> Interval<P> {
+    pub fn contains(&self, other: &P) -> bool {
+        self.lower <= *other && *other <= self.upper
+    }
+}
+
+impl<P: Sequence + PartialOrd + Clone> Interval<P> {
+    /// Returns up to two intervals that contain the range before and/or after given point,
+    /// neither of which contain the point.
+    pub fn split(&self, point: &P) -> Vec<Self> {
+        let mut ret = vec![];
+        let before = point.before();
+        let after = point.after();
+        if self.lower <= before {
```

Comment:
This will become a nicer match statement on the option returned by `before` once that's changed